pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java

```diff
@@ -274,26 +274,27 @@ public Map<String, String> getProperties(){
             return properties;
         }
     }
-    protected static class KeyValueKafkaRecord<V> extends KafkaRecord implements KVRecord<Object, Object> {
 
-        private final Schema<Object> keySchema;
-        private final Schema<Object> valueSchema;
+    protected static class KeyValueKafkaRecord<K, W> extends KafkaRecord implements KVRecord<K, W> {
```

Comment:
We could do
```java
protected static class KeyValueKafkaRecord<K, W> extends KafkaRecord<KeyValue<K, W>> implements KVRecord<K, W>
```
If we could change KVRecord to
```java
public interface KVRecord<K, V> extends Record<KeyValue<K, V>>
```
But KVRecord is marked as `@InterfaceAudience.Public` `@InterfaceStability.Stable` so I don't think it's possible to change it ?

Reply:
It is not possible to change it, in the past I tried by it has some consequences I can't remember well